source/regression1.md

```diff
@@ -596,23 +596,28 @@ on `sacr_gridsearch`. As we did in the {ref}`classification2` chapter,
 we will wrap the `cv_results_` output in a data frame, extract
 only the relevant columns, compute the standard error based on 5 folds, 
 and rename the parameter column to be more readable.
+In `scikit-learn`, it is easier to work with a data frame as the input features,
+rather than a series. So, when we select a single column,
+we explicitly convert the series to a data frame with the `to_frame()` method.
+The target variable can be a series, so we don't need to convert it to a data frame.
 
 ```{code-cell} ipython3
-# fit the GridSearchCV object 
+# fit the GridSearchCV object
 sacr_fit = sacr_gridsearch.fit(
-                  sacramento_train[["sqft"]],
-                  sacramento_train[["price"]]
-              )
-# retrieve the CV scores
-sacr_results = pd.DataFrame(sacr_fit.cv_results_)[
-    ["param_kneighborsregressor__n_neighbors", "mean_test_score", "std_test_score"]
-]
-sacr_results = sacr_results.assign(
-    sem_test_score = sacr_results["std_test_score"] / 5**(1/2)
-).rename(
-    columns = {"param_kneighborsregressor__n_neighbors" : "n_neighbors"}
-).drop(
-    columns = ["std_test_score"]
+    sacramento_train["sqft"].to_frame(),
```

Comment:
I think it is important to be explicit here. Previously we just used double bracket `[[...]]` with a single column name, which does return a dataframe, but it is not something we discuss anywhere in the book. We could add a discussion of this also and keep using double bracket for a single column name but `to_frame()` is a bit more versatile since it can be used in chains too and it might be confusing that we sometimes use single and sometimes double bracket with a single column

Reply:
@joelostblom rather than introducing a new function, why not just remind readers that `[...]` outputs a DF with a list of colns, and `["blah"]` is a list with a single element, so `[["blah"]]` outputs a df with one coln.

I think I might prefer that over introducing a new function here when there's a perfectly good tool the students already know

Reply:
I agree that we could change and explain the difference between extracting a single column with single or double brackets. The reason that I went with `to_frame` is because there is [a previous  worksheet](https://github.ubc.ca/UBC-DSCI/dsci-100-instructor/pull/797/files#r2574) where we need to use `to_frame` in  a method chain (where we can't use a single element list), so we need to explain that somewhere. If we go with the double brackets solution here, we could wrap the method chain in the worksheet in a call to  `pd.Dataframe` instead; it looks slightly less neat but not a huge deal.

After completing some of the machine learning worksheets, one more advantage of using `pd.Dataframe` over `to_frame` is that there are some dictionaries from `cv_results_` that we wrap in `pd.Dataframe` so using the same method for converting series to dataframes might make it easier to remember and they will not mix up when they can use `to_frame` and when they can't.

So all in all, I think I'm also leaning towards `pd.DataFrame` over `to_frame`. What do you think about preferring `pd.DataFrame` even over double brackets for a single column? So for the code above it would be `pd.DataFrame(sacramento_train["sqft"])`. Then things are very explicit and we always use the same method to convert a series to a dataframe, but the code is a bit inelegant and more tedious to write.

Reply:
The constructor call is definitely too bulky to be used everywhere. I think the best option is to go with the simplest one in each scenario (and properly introduce them the first time they're used, or a reminder/pointer if the introduction is far enough away and they aren't used regularly). My order of preference is:
1. `[[...]]`
2. `(...).to_frame()`
3. `pd.DataFrame(...)`

`[[....]]` handles basically every case we encounter. As long as we properly introduce this the first time it's done, it should be fine (since it will probably be seen somewhat frequently throughout). Then `to_frame` can get used where it's strictly necessary, with perhaps a reminder for the reader why we use it since it'll be somewhat infrequent. `pd.DataFrame` I imagine only is really needed for converting dictionaries, so again we can mention that there the one (?) time it happens.